docs/ref/forms/widgets.txt

```diff
@@ -157,6 +157,19 @@ this, you use the :attr:`Widget.attrs` argument when creating the widget::
         url = forms.URLField()
         comment = forms.CharField(widget=forms.TextInput(attrs={'size': '40'}))
 
+Alternatively, you can make these modifications directly at the point of instantiation
+using the form.fields attribute::
+
+    class CommentForm(forms.Form):
+        name = forms.CharField()
+        url = forms.URLField()
+        comment = forms.CharField()
+
+        def __init__(self, *args, **kwargs):
+            super(CommentForm, self).__init__(*args, **kwargs)
+            self.fields['name'].widget.attrs = {'class': 'special'}
```

Comment:
This isn't the best example of why you would override ``__init__`` to do this. The attributes being set are invariant in this example. (The example referred to by @timgraham in the ticket description is performing some formatting of the label and help text to use as a placeholder.) Perhaps make them based on something passed into the form?

If the intent of this was to show how to avoid specifying the widget explicitly, using the default for the field, then the following also works and is perfectly legal from my experience:

```python
class CommentForm(forms.Form):
    name = forms.CharField()
    url = forms.URLField()
    comment = forms.CharField()

    name.widget.attrs = {'class': 'special'}
    comment.widget.attrs.update(size='40')
```

Note that I have shown here that it is possible to override the attributes entirely or just update some of them.